gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -296,9 +300,56 @@ def __init__(self,
               self._process_header_lines),
           **kwargs)
 
+    if isinstance(self, PySamParser):
+      # In an absence of "#CHROM POS..." header line, PySam process would lock.
+      # However, we need to support cases when no records in files were provided
+      # even if the above header line is missing. To circumvent the error, read
+      # the file until the first line is found, and verify validity.
+      self._header_lines = None
+      if isinstance(range_tracker, range_trackers.UnsplittableRangeTracker):
+        secondary_tracker = range_trackers.UnsplittableRangeTracker(
+            copy(range_tracker._range_tracker))
+      else:
+        secondary_tracker = copy(range_tracker)
+      self._verify_header(copy(text_source), secondary_tracker)
+
     self._text_lines = text_source.read_records(self._file_name,
                                                 range_tracker)
 
+  def after_read(self):
+    return
+
+  def _verify_header(self, text_source, range_tracker):
+    try:
+      text_lines = text_source.read_records(self._file_name,
+                                            range_tracker)
+
+      text_line = next(text_lines).strip()
+      while not text_line.strip():  # skip empty lines.
+        # This natively raises StopIteration if end of file is reached.
+        text_line = next(text_lines).strip()
+      if not self._header_lines[-1].startswith(LAST_HEADER_LINE_PREFIX):
+        raise ValueError('Header missing CHROM line.')
+      header = libcbcf.VariantHeader()
+      for header_line in self._header_lines[:-1]:
+        header.add_line(header_line)
+
```

Comment:
It would be nice to describe the checks here and why they exist (presumably these are things you found by trial and error than htslib doesn't like?)